### PR TITLE
Feat✨ (#8161) Implement logic to fetch report data part-2

### LINKED ISSUE #8161: Implement logic to fetch report data

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Fixes #8161 
### Describe your changes :

I worked on implementing logic to fetch report data on the data insight page.
This PR includes logic to fetch these chart data.
- mostActiveUsers
- mostViewedEntities
- pageViewsByEntities
- dailyActiveUsers

- [x] New feature

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

**Review by ShaileshParmar11 — APPROVED** [review]
UI changes looks good!

**Review by ShaileshParmar11 — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/interface/data-insight.interface.ts [hunk 1]
```diff
@@ -27,3 +30,7 @@ export interface ChartFilter {
   startTs: number;
   endTs: number;
 }
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface DataInsightChartTooltipProps extends TooltipProps<any, any> {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/interface/data-insight.interface.ts:34]: why `any`

File: openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts [hunk 2]
```diff
@@ -394,3 +402,10 @@ export const formatDateTimeFromSeconds = (date: number) => {
 
   return dateTime.toLocaleString(DateTime.DATETIME_MED);
 };
+
+/**
+ *
+ * @returns Current date timestamp
+ */
+export const getCurrentDateTimeStampInMilliSeconds = () =>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:410]:
```suggestion
export const getCurrentDateStampInMilliSeconds = () =>
```

It's not returning time
Can we stick to mm-dd-yyyy format? and don't make too many formatting methods

File: openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts [hunk 3]
```diff
@@ -336,6 +337,13 @@ export const getFormattedDateFromMilliSeconds = (
 export const getDateTimeFromMilliSeconds = (timeStamp: number) =>
   DateTime.fromMillis(timeStamp).toLocaleString(DateTime.DATETIME_MED);
 
+/**
+ * @param seconds EPOCH seconds
+ * @returns Formatted duration for valid input. Format: 00:09:31
+ */
+export const getDurationFromSeconds = (seconds: number) =>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:344]:
```suggestion
export const getTimeDurationFromSeconds = (seconds: number) =>
```

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/interface/data-insight.interface.ts (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/interface/data-insight.interface.ts:34]: we don't have a type for Tooltip value and Tooltip that's why we have to use the type "any".

File: openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts (hunk 2 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:410]: @chirag-madlani , This method uses another `getTimeStampByDate` which requires date format 'yyyy-MM-dd', which will return the timestamp for the same.

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/utils/TimeUtils.ts:410]: As per the discussion, I have removed the `getCurrentDateTimeStampInMilliSeconds` method as we are not using it and marked `getTimeStampByDate` as deprecated.

**Review by chirag-madlani — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

File: ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py [hunk 4]
```diff
@@ -239,7 +239,7 @@ def _get_user_details(self, user_id: str) -> Optional[dict]:
         )
 
         if not user_entity:
-            return None
+            return {}
```
Comment (pmbrull) [inline ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py:242]: is this really needed? can be more error prone than reviewing a null value cc @TeddyCr

**Comment by harshach** [comment]
@Sachin-chaurasiya lets make sure there is no "Entity Name" anywhere and let's call them as Data Asset. Where we are providing these names there should be a link where users can click and go to the details page

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya lets make sure there is no "Entity Name" anywhere and let's call them as Data Asset. Where we are providing these names there should be a link where users can click and go to the details page

Sure, @harshach, I will make the changes, regarding the making data asset link is clickable we will need href URL from backend cc : @TeddyCr

**Review by TeddyCr — APPROVED** [review]

File: ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py (hunk 4 above)
Comment (TeddyCr) [reply ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py:242]: yes, pass a dict to the user id key we get from the event data and retrieve those details to process it down below.

**Comment by TeddyCr** [comment]
@harshach @Sachin-chaurasiya issue for reference https://github.com/open-metadata/OpenMetadata/issues/8638

**Pull request merged**